tests/unit/jobserver/test_forms.py

```diff
@@ -104,27 +104,42 @@ def test_jobrequestcreateform_with_bad_codelists(
     ],
 )
 def test_workspacecreateform_success(name, cleaned_name):
+    """
+    When the form is instantiated with valid data then:
+        * The form validates.
+        * The data ends up in the cleaned_data.
+        * The name field is cleaned by converting to lower case.
+    """
     project = ProjectFactory()
-    data = {
-        "name": name,
-        "repo": "http://example.com/derp/test-repo",
-        "branch": "test-branch",
-        "purpose": "test",
-    }
     repos_with_branches = [
         {
             "name": "test-repo",
             "url": "http://example.com/derp/test-repo",
             "branches": ["test-branch"],
-        }
+        },
+        {
+            "name": "test-repo2",
+            "url": "http://example.com/derp/test-repo2",
+            "branches": ["test-branch2"],
+        },
     ]
-    form = WorkspaceCreateForm(project, repos_with_branches, data)
+    data = {
+        "name": name,
+        "repo": "http://example.com/derp/test-repo2",
+        "branch": "test-branch2",
+        "purpose": "test purpose",
+    }
+    form = WorkspaceCreateForm(project, repos_with_branches, data=data)
 
     assert form.is_valid()
     assert form.cleaned_data["name"] == cleaned_name
+    assert form.cleaned_data["repo"] == "http://example.com/derp/test-repo2"
+    assert form.cleaned_data["branch"] == "test-branch2"
+    assert form.cleaned_data["purpose"] == "test purpose"
 
 
 def test_workspacecreateform_unknown_branch():
+    """When the form cleaned_data has an unknown branch, validation fails."""
```

Comment:
Here, and below, I feel we should incorporate the comments into the names of the functions. Doing so would make pytest's output clearer - if a test failed, then we wouldn't have to cross-reference with the function's comment to learn more about why the test failed.

Reply:
I'd be happy to add `_validation_fails` to the end of each. I think the docstrings are more legible when viewing the code and barely make the functions longer, so can we do both?